### PR TITLE
feat(core): add jump to block for cmdk

### PR DESCRIPTION
**Pull request by JimmFly**
work for #4591 

https://github.com/toeverything/AFFiNE/assets/102217452/90af6112-e74f-4da9-99ac-7a3f1596523a

### CODE REVIEW

File: packages/frontend/core/src/components/page-detail-editor.tsx [hunk 1]
```diff
@@ -80,6 +86,20 @@ const EditorWrapper = memo(function EditorWrapper({
     return fontStyle.value;
   }, [appSettings.fontStyle]);
 
+  const query = useRouterQuery();
+  const blockId = query.get('blockId');
```
Comment (joooye34) [inline packages/frontend/core/src/components/page-detail-editor.tsx:90]: why is query? hash is defined for this

File: packages/frontend/core/src/components/page-detail-editor.tsx [hunk 2]
```diff
@@ -80,6 +86,20 @@ const EditorWrapper = memo(function EditorWrapper({
...
+
+  if (blockId) {
+    const blockElement = document.querySelector(`[data-block-id="${blockId}"]`);
+    if (blockElement) {
+      blockElement.scrollIntoView({
```
Comment (joooye34) [inline packages/frontend/core/src/components/page-detail-editor.tsx:95]: this effect logic should in useEffect

File: packages/frontend/core/src/components/pure/cmdk/data.tsx [hunk 3]
```diff
@@ -205,17 +218,23 @@ export const usePageCommands = () => {
       });
     } else {
       // queried pages that has matched contents
-      const searchResults = Array.from(
-        workspace.blockSuiteWorkspace.search({ query }).values()
-      ) as unknown as { space: string; content: string }[];
+      const searchResults = workspace.blockSuiteWorkspace.search({
```
Comment (pengx17) [inline packages/frontend/core/src/components/pure/cmdk/data.tsx:221]: it reminds me that we shall have a debounce for global search here. 🤔

File: packages/frontend/core/src/components/pure/cmdk/data.tsx [hunk 4]
```diff
@@ -205,17 +218,23 @@ export const usePageCommands = () => {
...
+        query,
+      }) as unknown as Map<string, SearchResultsValue>;
+      const resultValues = Array.from(searchResults.values());
+      // const blockIds = searchResults.keys();
```
Comment (pengx17) [inline packages/frontend/core/src/components/pure/cmdk/data.tsx:225]: remove outdated comments

File: packages/frontend/core/src/components/page-detail-editor.tsx [hunk 5]
```diff
@@ -80,6 +87,29 @@ const EditorWrapper = memo(function EditorWrapper({
...
+  const blockElement = useMemo(() => {
+    if (!blockId || loading) {
+      return null;
+    }
+    return document.querySelector(`[data-block-id="${blockId}"]`);
+  }, [blockId, loading]);
+
+  useEffect(() => {
+    if (blockElement) {
+      setTimeout(
```
Comment (joooye34) [inline packages/frontend/core/src/components/page-detail-editor.tsx:101]: it is ok to do effect action here, seems timeout is unneccessary.

Comment (JimmFly) [reply packages/frontend/core/src/components/page-detail-editor.tsx:101]: In order to ensure that scrolling can occur after the dom is rendered correctly

**Review by joooye34 — APPROVED** [review]

**Pull request merged**